include/seqan3/io/sequence_file/format_sam.hpp

```diff
@@ -0,0 +1,347 @@
+   *
+   * This implementation supports the following optional features of the format:
+   *
+   * line breaks and/or other whitespace characters in any part of the sequence (only when reading!), optional tags
+   * are going to be ignored (when reading)
+   *
+   */
+class sequence_file_format_sam
+{
+public:
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+     //!\brief Default constructor is explicitly deleted, you need to give a stream or file name.
+    sequence_file_format_sam() = default;
+    //!\brief Copy construction is explicitly deleted, because you can't have multiple access to the same file.
+    sequence_file_format_sam(sequence_file_format_sam const &) = delete;
+    //!\brief Copy assignment is explicitly deleted, because you can't have multiple access to the same file.
+    sequence_file_format_sam & operator=(sequence_file_format_sam const &) = delete;
+    //!\brief Move construction is defaulted.
+    sequence_file_format_sam(sequence_file_format_sam &&) = default;
+    //!\brief Move assignment is defaulted.
+    sequence_file_format_sam & operator=(sequence_file_format_sam &&) = default;
+    //!\}
+
+    //!\brief The valid file extensions for this format; note that you can modify this value.
+    static inline std::vector<std::string> file_extensions
+    {
+        { "sam" },
+    };
+
+    //!\copydoc sequence_file_input_format_concept::read
+    template <typename stream_type,     // constraints checked by file
+              typename seq_legal_alph_type, bool seq_qual_combined,
+              typename seq_type,        // other constraints checked inside function
+              typename id_type,
+              typename qual_type>
+    void read(stream_type                     & stream,
+              sequence_file_input_options<seq_legal_alph_type, seq_qual_combined> const & options,
+              seq_type                        & sequence,
+              id_type                         & id,
+              qual_type                       & qualities)
+    {
+        auto stream_view = view::subrange<decltype(std::istreambuf_iterator<char>{stream}),
+                                          decltype(std::istreambuf_iterator<char>{})>
+                            {std::istreambuf_iterator<char>{stream},
+                             std::istreambuf_iterator<char>{}};
+
+        auto stream_it = ranges::begin(stream_view);
+
+        // cache the begin position so we write quals to the same position as seq in seq_qual case
+        size_t sequence_size_before = 0;
+        size_t sequence_size_after = 0;
+        if constexpr (!detail::decays_to_ignore_v<seq_type>)
+            sequence_size_before = ranges::size(sequence);
+
+        auto const is_tab = is_char<'\t'>;
+        auto const is_free = is_char<'\r'>;
+
+        while (is_char<'@'>(*stream_it))
+        {
+            detail::consume(stream_view | view::take_line);
+            stream_it = ranges::begin(stream_view);
```

Comment:
why do you need to set the stream_it here while in line 186 (a similar loop) you do not?

Reply:
This is only necessary for g++7, in g++8 I would not need that line at all and I think the reason I need it here in g++7 but not in the line below is because here I jump over an entire line and need to keep stream_it updated, below I only jump over one letter at a time and adapt stream_it via ++stream_it.

Reply:
then don't use the stream_it at all but directly ut `ranges::begin(stream_view)` into the while condition: `while (is_char<'@'>(*ranges::begin(stream_view)))`